lib/engine/game/g_1824_cisleithania/game.rb

```diff
@@ -453,25 +459,37 @@ def select_randomly(collection)
         def stackify(company, header)
           return header unless company.stack
 
-          real_header = "#{header} STACK #{company.stack}"
-          real_header += ' (CR)' if company.stack == 1
-          real_header
+          case company.stack
+          when 1
+            'Construction Railway (Stack 1)'
+          when 2
+            'Large Pre-staatsbahn (Stack 2)'
+          when 3
+            'Small Pre-staatsbahn (Stack 3)'
+          when 4
+            'Coal Railway (Stack 4)'
+          end
```

Comment:
Style suggestion: very simple `case` statements like this can be more readable if you use the single line format for the `when` tests.
```suggestion
          case company.stack
          when 1 then 'Construction Railway (Stack 1)'
          when 2 then 'Large Pre-staatsbahn (Stack 2)'
          when 3 then 'Small Pre-staatsbahn (Stack 3)'
          when 4 then 'Coal Railway (Stack 4)'
          end
```